### ISSUE TITLE
`multipartBody` writing `Content-Type` header twice

### ISSUE THREAD

**Issue by guersam** [softwaremill/tapir#184]
### Data structure

```scala
@JsonCodec
case class UserProfile(name: String, email: Email)

case class UserInfo(profile: UserProfile, image: Part[File])
```

### Expected body
```
-----boundary
Content-Disposition: form-data; name="profile"
Content-Type: application/json

{ "name": "guersam", "email": "guersam@guersam.com" }
-----boundary
Content-Disposition: form-data; name="image"; filename="a.jpg"
Content-Type: image/jpeg

...
```

### Actual body
```
-----boundary
Content-Disposition: form-data; name="profile"
Content-Type: text/plain; charset=utf-8
Content-Type: application/json

...
```

**Comment by adamw** [comment]
I couldn't reproduce the double-header, but I fixed setting content-type basing on the codec in 0.9.1. Please test & reopen if still doesn't work

**Comment by guersam** [comment]
My bug report was incompleted because it happened when I manually added `Content-Type` header to the `Part` fields. Confirmed that it's fixed by bc592bf5 as below:

0.9.0
```scala
      Multipart(
        "image",
        FileBody(
          com.softwaremill.sttp.internal.SttpFileCompanionExtensions$$anon$1@14f232c4,
          Some("application/octet-stream")
        ),
        Some("myfile"),
        Some("application/octet-stream"),
        Map("Content-Type" -> "image/jpeg")
      )
```

0.9.1
```scala
      Multipart(
        "image",
        FileBody(
          com.softwaremill.sttp.internal.SttpFileCompanionExtensions$$anon$1@5dfb65d5,
          Some("application/octet-stream")
        ),
        Some("myfile"),
        Some("image/jpeg"),
        Map()
      )
```

**Comment by adamw** [comment]
Released in 0.12.7
